@xen-orchestra/backups/writers/_AbstractWriter.js

```diff
@@ -0,0 +1,31 @@
+const { asyncMap } = require('@xen-orchestra/async-map')
+const { createLogger } = require('@xen-orchestra/log')
+
+const { BACKUP_DIR } = require('../_getVmBackupDir')
+
+const { debug } = createLogger('xo:backups:AbstractWriter')
+
+exports.AbstractWriter = class AbstractWriter {
+  constructor({ adapter } = {}) {
+    this._adapter = adapter
+  }
+
+  async cleanCorruptedBackups() {
+    const adapter = this._adapter
+    if (adapter !== undefined) {
+      await asyncMap(await adapter.handler.list(BACKUP_DIR, { prependDir: true }), async vmDir =>
+        adapter.cleanVm(vmDir, { remove: true, merge: true, onLog: debug })
+      )
+    }
```

Comment:
Cannot be there, not generic enough.

Reply:
What's about adding an `AbstractExportWriter` class which implement this method ?

Reply:
How?

JavaScript does not support multiple inheritance.

Reply:
Right, it can be a mixin which will be added to the class prototype.